hibernate-core/src/main/java/org/hibernate/type/OneToOneType.java

```diff
@@ -114,11 +114,6 @@ public void nullSafeSet(PreparedStatement st, Object value, int index, boolean[]
 	}
 
```

Comment:
The lines that precede the following deletion is:
```
 	@Override
 	public void nullSafeSet(PreparedStatement st, Object value, int index, boolean[] settable, SharedSessionContractImplementor session) {
 		//nothing to do
  	}
```
IOW, when ``` OneToOneType#nullSafeSet(PreparedStatement, Object, int, boolean[], SharedSessionContractImplementor)``` is called, nothing is done.

When ```OneToOneType#nullSafeSet(PreparedStatement st, Object, int, SharedSessionContractImplementor)``` is called, ```EntityType#nullSafeSet(PreparedStatement st, Object, int, SharedSessionContractImplementor) will actually be called, and a value will be bound.

I don't think it makes sense to bind a value in one case, but not another. I would guess that something will break somewhere.

I'll try to come up with a test case.

Reply:
I think this is OK. 

```OneToOneType#nullSafeSet(PreparedStatement, Object, int, boolean[], SharedSessionContractImplementor)``` is used when inserting, updating, or deleting the entity with the one-to-one association. In those cases,  ```OneToOneType#nullSafeSet``` should do nothing.

Reply:
I added a comment that I checked this out and it was OK. I think you can see the comment when you look at the "Files Changed" tab.